inc/connector/namespace.php

```diff
@@ -11,6 +14,354 @@
+	$job->interval = $event->interval;
+	$job->schedule = $event->schedule;
+	$job->save();
+
+	// Rescheduled.
+	return true;
+}
+
+/**
+ * Unschedule a previously scheduled event.
+ *
+ * The $timestamp and $hook parameters are required so that the event can be
+ * identified.
+ *
+ * @param null|bool $pre       Value to return instead. Default null to continue unscheduling the event.
+ * @param int       $timestamp Timestamp for when to run the event.
+ * @param string    $hook      Action hook, the execution of which will be unscheduled.
+ * @param array     $args      Arguments to pass to the hook's callback function.
+ * @return null|bool True if event successfully scheduled. False for failure.
+ */
+function pre_unschedule_event( $pre, $timestamp, $hook, $args ) {
+	// First check if the job exists already.
+	$job = Job::get_jobs_by_query(
+		[
+			'hook' => $hook,
+			'timestamp' => $timestamp,
+			'args' => $args,
+		]
+	);
+
+	if ( empty( $job ) || empty ( reset( $job ) ) ) {
+		// The job does not exist.
+		return false;
+	}
+
+	// Delete it.
+	reset( $job )->delete();
+
+	return true;
+}
+
+/**
+ * Unschedules all events attached to the hook with the specified arguments.
+ *
+ * Warning: This function may return Boolean FALSE, but may also return a non-Boolean
+ * value which evaluates to FALSE. For information about casting to booleans see the
+ * {@link https://php.net/manual/en/language.types.boolean.php PHP documentation}. Use
+ * the `===` operator for testing the return value of this function.
+ *
+ * @param null|array $pre  Value to return instead. Default null to continue unscheduling the event.
+ * @param string     $hook Action hook, the execution of which will be unscheduled.
+ * @param array|null $args Arguments to pass to the hook's callback function, null to clear all
+ *                         events regardless of arugments.
+ * @return bool|int  On success an integer indicating number of events unscheduled (0 indicates no
+ *                   events were registered with the hook and arguments combination), false if
+ *                   unscheduling one or more events fail.
+*/
+function pre_clear_scheduled_hook( $pre, $hook, $args ) {
+	// First check if the job exists already.
+	$jobs = Job::get_jobs_by_query(
+		[
+			'hook' => $hook,
+			'args' => $args,
+			'limit' => 100,
+		]
+	);
+
+	if ( empty( $jobs ) ) {
+		// No jobs to unschedule.
+		return 0;
+	}
+
+	$ids = wp_list_pluck( $jobs, 'id' );
+
+	global $wpdb;
+
+	// Clear all scheduled events for this site
+	$table = Job::get_table();
+
+	$sql = "DELETE FROM `{$table}` WHERE site = %d";
+	$sql_params[] = get_current_blog_id();
+
+	$sql .= ' AND id IN(' . implode( ',', array_fill( 0, count( $ids ), '%d' ) ) . ')';
+	$sql_params = array_merge( $sql_params, $ids );
+
+	$query = $wpdb->prepare( $sql, $sql_params );
+	$results = $wpdb->query( $query );
+
+	// Flush the caches.
+	Job::flush_query_cache();
+	array_walk( $ids, function( $id ) {
```

Comment:
Why is this `array_walk`? Could it be `array_map` instead?

Reply:
Also, needs a space after `function`

Reply:
Syntactically `array_map()` is for transforming and returning a modified array, this could just as easily be a foreach though.